sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -794,7 +859,15 @@ private String getApplicationName() {
   public User getUser() {
     User user = new User();
 
-    user.setId(getAndroidId());
+    try {
+      Object androidId = contextData.get().get("androidId");
+
+      if (androidId != null) {
+        user.setId((String) androidId);
```

Comment:
We've been discussing adding this also to `context.device.id` which is not in the protocol yet but could be align the approach with iOS  since neither have a consistent way to give an id to the device.
Not necessarily for this PR though. /cc @fetzig